sympy/printing/latex.py

```diff
@@ -767,10 +768,21 @@ def _print_UndefinedFunction(self, expr):
         return self._hprint_Function(str(expr))
 
     def _print_FunctionClass(self, expr):
-        if hasattr(expr, '_latex_no_arg'):
-            return expr._latex_no_arg(self)
-
-        return self._hprint_Function(str(expr))
+        from sympy.functions.special.tensor_functions import KroneckerDelta
+        from sympy.functions.special.gamma_functions import gamma, lowergamma
+        from sympy.functions.special.beta_functions import beta
+        from sympy.functions.special.delta_functions import DiracDelta
+        from sympy.functions.special.error_functions import Chi
+        special_cases = {KroneckerDelta: r'\delta',
+                         gamma:  r'\Gamma',
+                         lowergamma: r'\gamma',
+                         beta: r'\operatorname{B}',
+                         DiracDelta: r'\delta',
+                         Chi: r'\operatorname{Chi}'}
+        if expr in special_cases:
```

Comment:
Should probably do a subclass check. Although if the subclass name is different it should print the actual class name.

Reply:
I'm weak in subclassing; do you have an example in mind of the kind of things to check for? This is within `_print_FunctionClass` so if the printer got here, that means `type(expr).__mro__` did not have any subclass with a custom _print_ implemented. If someone subclasses, say, DiracDelta and doesn't add a _print_ for their new class, it will print as `r'\delta'.

Reply:
I now see what you mean. Added a commit replacing direct comparison by `issubclass` (both in latex and pretty) and added tests.

Reply:
I mean like

```py
class beta(sympy.beta):
    ...
```

should still print as `\beta`, but

```py
class mybeta(sympy.beta):
    ...
```

shouldn't (because the name changed to `mybeta`).

Reply:
I see. That's easy to make work for unapplied (FunctionClass) but to make `mybeta(x, y)` behave in the same way I'll need to remove all the _print_beta, _print_gamma, etc moving them to some special_function dictionary used by _print_Function. Otherwise, the driver following `(mybeta, beta, Function,..)` hierarchy will dispatch mybeta to _print_beta, not to _print_Function.

But it still won't work well because customizations are not just name changes; say, KroneckerDelta has its arguments placed as subscripts. Should this argument-placement behavior be inherited by subclasses that have a different name? Unclear to me. 

Considering all this, I think I prefer the original approach without subclass check: no special printing rules pass to subclasses.

Reply:
The current behavior of functions is that notation passes to subclasses, regardless of how they are named.
```
class myg(gamma):
  pass
latex(myg(x))    r'\Gamma{\left(x \right)}'
```
To change the printed name, one would have to add `_print_myg` method to each printer. To my understanding, this is the trade-off that was [consciously made](http://docs.sympy.org/dev/modules/printing.html):

> works great for built-in sympy objects, but not that good for user defined classes where it is inconvenient to patch the printers.

Reply:
I think if we want to change this notation inheritance, the most practical way is to truncate the MRO list in printer.py, similar to how it was done for undefined functions. The rule would be: when MRO contains `Function`, ignore the subclasses of Function except for the smallest one. So instead of `(mybeta, beta, Function,..)` the list is `(mybeta, Function,..)` and since there is no _print_mybeta, it gets dispatched to _print_Function which looks at the function name. Does this sound reasonable?